src/sentry/integrations/jira_server/integration.py

```diff
@@ -132,8 +138,70 @@ def dispatch(self, request, pipeline):
         )
 
 
+class OAuthLoginView(PipelineView):
+    """
+    Start the OAuth dance by creating a request token
+    and redirecting the user to approve it.
+    """
+    @csrf_exempt
+    def dispatch(self, request, pipeline):
+        if 'oauth_token' in request.GET:
```

Comment:
Does the user have an accept/reject button to press? What happens here if the user doesn't authorize us?

Reply:
The user does have a deny/allow button inside Jira. If they click deny, they are sent back to us with an `oauth_token` in the GET params. When we try to fetch the access token we get a 401 error.

Right now we display 'Could not fetch an access token from Jira' in the pipeline error page.

Reply:
That's fine. The user took a direct action so there shouldn't be confusion there.